File: solutions/r/raindrops/1/raindrops.R
raindrops <- function(number) {
  result <- ""
  if (number %% 3 == 0) {
    x1 <- "Pling"
  } else {
    x1 <- ""
  }
  if (number %% 5 == 0) {
    x2 <- "Plang"
  } else {
    x2 <- ""
  }
  if (number %% 7 == 0) {
    x3 <- "Plong"
  } else {
    x3 <- ""
  }
  result <- paste(x1, x2, x3, sep="")
  if (result == "") {
    return (as.character(number))
  } else {
    return (result)
  }
}

File: solutions/ruby/raindrops/1/raindrops.rb
=begin
Write your code for the 'Raindrops' exercise in this file. Make the tests in
`raindrops_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/raindrops` directory.
=end

class Raindrops
  
  def self.convert(num)
    result = ""
    if num % 3 == 0 then
      result += "Pling"
    end
    if num % 5 == 0 then
      result += "Plang"
    end
    if num % 7 == 0 then
      result += "Plong"
    end

    if result == "" then
      "#{num}"
    else
      result
    end
  end

end
